bpo-34616: add flags to allow top-level-await (#13148)
python/cpython: Doc/library/functions.rst

   can be found as the :attr:`~__future__._Feature.compiler_flag` attribute on
   the :class:`~__future__._Feature` instance in the :mod:`__future__` module.

>    The optional argument *flags* also control wether the compiled source is

Reviewer: ```suggestion
   The optional argument *flags* also controls whether the compiled source is
```

   allowed to contain top-level ``await``, ``async for`` and ``async with``.

>    When the bit ``ast.PyCF_ALLOW_TOP_LEVEL_AWAIT`` is set, the the return code

Author: Note to self. Double "the" here.

>    object has ``CO_COROUTINE`` set in ``co_code``, and can be interacitvely

Reviewer: ```suggestion
   object has ``CO_COROUTINE`` set in ``co_code``, and can be interactively
```

>    executed via ``yield from eval(code_object)``.

Reviewer 2: ```suggestion
   executed via ``await eval(code_object)``.
```

      Previously, :exc:`TypeError` was raised when null bytes were encountered
      in *source*.

   .. versionadded:: 3.8

>       *flags* can now be used to accept source that would contain a top-level

Reviewer 2: ``ast.PyCF_ALLOW_TOP_LEVEL_AWAIT`` can now be passed in *flags* to enable support
for top-level await, async for, and async with.

Author: Done.

>       support for top-level await, async for, and async with.

Author: note to self. Put `async ...` between backquotes.
